ibis/expr/datatypes/core.py

```diff
@@ -9,68 +9,49 @@
 from numbers import Integral, Real
 from typing import Any, Iterable, Literal, NamedTuple, Optional
 
-import numpy as np
-import pandas as pd
 import toolz
-from multipledispatch import Dispatcher
 from public import public
-from typing_extensions import get_args, get_origin, get_type_hints
+from typing_extensions import Self, get_args, get_origin, get_type_hints
```

Comment:
Isn't the `Self` type only available since 3.11? Or is that not an issue because you're pulling it from `typing_extensions`?

Reply:
Correct.